### PR TITLE
fix: Language review and Vale rules update

### PR DESCRIPTION
**Pull request by themr0c**
Adapt vale configuration:

* Skip validation on metadata (`:attribute:` and `id` statements).
* Use the  new `CheDocs.TechnicalTerms` rule  to define capitalization rules, with severity `error`, instead of the Vocab (`accept.txt`, `reject.txt`).
* Use the Vocab (`accept.txt`, `reject.txt`) only to expand the dictionary (words triggering `Spelling` errors).
* Raise the severity of the `CheDocs.Attributes` substitution rule.

Fix the content.

### CODE REVIEW

File: .vale/styles/Vocab/Che/accept.txt [hunk 1]
```diff
@@ -37,7 +37,6 @@ AsciiDoc
 autostart
 AWS|aws
 Bitbucket
-boolean
```
Comment (yhontyk) [inline .vale/styles/Vocab/Che/accept.txt:40]: Why do we remove this? For different purposes, Boolean and boolean are accepted.

**Comment by yhontyk** [comment]
>26:19  error       Use 'URL' instead of 'url'      CheDocs.TechnicalTerms 
 >26:19  error       Use 'URL' instead of 'url'.     Vale.Terms       

Duplicating rule

File: .vale/styles/Vocab/Che/reject.txt [hunk 2]
```diff
@@ -12,8 +13,6 @@ now
 operator
 our
 out of the box
-[Pp]lease
-pod
```
Comment (yhontyk) [inline .vale/styles/Vocab/Che/reject.txt:16]: `pod` will be reported by Vale as an unknown work AFAIK.

**Comment by yhontyk** [comment]
@themr0c can we just merge this PR https://github.com/eclipse/che-docs/pull/1888 and rebase on it? From what I see, there are only two attributes that need to be fixed in that one and it's ready to go.

File: .vale/styles/Vocab/Che/accept.txt (hunk 1 above)
Comment (themr0c) [reply .vale/styles/Vocab/Che/accept.txt:40]: Indeed. We have `[Bb]oolean` in the accept.txt (line 2) to add it to the dictionary (`Spelling` rules) and accept both forms. The line containing `boolean` is therefore obsolete.

File: .vale/styles/Vocab/Che/reject.txt (hunk 2 above)
Comment (themr0c) [reply .vale/styles/Vocab/Che/reject.txt:16]: This is a change in the `reject.txt` file which was refusing `pod`. The scoping for the vocabulary is not perfect, so for capitalization, the job is moving to the `TechnicalTerms` rule (with a more complete error message).

File: .vale/styles/Vocab/Che/accept.txt (hunk 1 above)
Comment (yhontyk) [reply .vale/styles/Vocab/Che/accept.txt:40]: Ah! Good! Sorry for missing it

File: .vale/styles/Vocab/Che/reject.txt (hunk 2 above)
Comment (yhontyk) [reply .vale/styles/Vocab/Che/reject.txt:16]: Ack, makes sense

File: modules/installation-guide/partials/proc_preparing-the-aws-system-for-installing-che.adoc [hunk 3]
```diff
@@ -4,6 +4,8 @@
 The following sections describe how to Configure {kubernetes} for {prod}
 on Amazon Elastic Compute Cloud (Amazon EC2).
 
+boolean Boolean
```
Comment (yhontyk) [inline modules/installation-guide/partials/proc_preparing-the-aws-system-for-installing-che.adoc]: This should not be here

File: .vale/styles/CheDocs/TechnicalTerms.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  '\sconfig maps': ConfigMaps
+  '\sDotnet': .NET
+  '\sdotnet': .NET
+  '\sgithub': GitHub
+  '\side': IDE
+  '\sjetbrains': Jetbrains
+  '\sjava': Java
+  '\netcoredebugoutput': NetcoredebugOutput
+  '\soperator': Operator
+  '\spod': Pod
```
Comment (yhontyk) [inline .vale/styles/CheDocs/TechnicalTerms.yml:20]: Do we capitalize it to match Kubernetes' documentation?
OpenShift has given up the k8s-style capitalization (they only capitalize Operators now).

File: .vale/styles/CheDocs/TechnicalTerms.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+extends: substitution
+ignorecase: false
+level: error
+message: Use '%s' instead of '%s'
+scope: sentence
+swap:
+  '\sContainer Registry': Red Hat Ecosystem Catalog
+  '\sContainer Catalog': Red Hat Ecosystem Catalog
+  '\sconfig map': ConfigMap
+  '\sconfig maps': ConfigMaps
```
Comment (yhontyk) [inline .vale/styles/CheDocs/TechnicalTerms.yml:11]: This is also according to k8s docs style but not OpenShift. OpenShift docs do not pluralize API objects https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#object-references

I'm mostly bringing this up so we as a team decide clearly on the style we want to follow and enforce in the docs.

**Comment by yhontyk** [comment]
@themr0c while running locally, I also encountered a duplicating rule

```
 81:21  error  Use 'Operator' instead of       Vale.Terms 
               'operator'.                                
 81:21  error  Avoid using 'operator'.         Vale.Avoid 
```

File: modules/installation-guide/partials/proc_preparing-the-aws-system-for-installing-che.adoc [hunk 6]
```diff
@@ -166,7 +168,7 @@ Output similar to the following means a needed role must be created manually:
 +
 [subs="+quotes",options="nowrap"]
 ----
-arn:aws:sts::269287474311:assumed-role...4bff is not authorized to perform: iam:CreateServiceLinkedRole on resource: arn:aws:iam::269287474311:role/aws-service-role/elasticloadbalancing.amazonaws.com/AWSServiceRoleForElasticLoadBalancing
+arn:aws:`ts::269287474311:assumed-role...4bff is not authorized to perform: iam:CreateServiceLinkedRole on resource: arn:aws:iam::269287474311:role/aws-service-role/elasticloadbalancing.amazonaws.com/AWSServiceRoleForElasticLoadBalancing
```
Comment (skabashnyuk) [inline modules/installation-guide/partials/proc_preparing-the-aws-system-for-installing-che.adoc]: is there a reason for this change?

**Review by skabashnyuk — APPROVED** [review]

File: .vale/styles/CheDocs/TechnicalTerms.yml (hunk 4 above)
Comment (themr0c) [reply .vale/styles/CheDocs/TechnicalTerms.yml:20]: I guess so.

File: modules/installation-guide/partials/proc_preparing-the-aws-system-for-installing-che.adoc (hunk 3 above)
Comment (themr0c) [inline modules/installation-guide/partials/proc_preparing-the-aws-system-for-installing-che.adoc]:
```suggestion
```

OMG. a debug leftover oO.

File: modules/installation-guide/partials/proc_preparing-the-aws-system-for-installing-che.adoc (hunk 6 above)
Comment (themr0c) [inline modules/installation-guide/partials/proc_preparing-the-aws-system-for-installing-che.adoc]:
```suggestion
arn:aws:sts::269287474311:assumed-role...4bff is not authorized to perform: iam:CreateServiceLinkedRole on resource: arn:aws:iam::269287474311:role/aws-service-role/elasticloadbalancing.amazonaws.com/AWSServiceRoleForElasticLoadBalancing
```

typo

**Review by yhontyk — APPROVED** [review]

**Comment by themr0c** [comment]
> @themr0c can we just merge this PR #1888 and rebase on it? From what I see, there are only two attributes that need to be fixed in that one and it's ready to go.

In #1888 I made the mistake to commit stuff in `modules/installation-guide/examples/system-variables.adoc`, I would prefer run things the other way around and deprecate #1888. For the `system-variables.adoc`, changes should go to the `che` repository. See: https://github.com/eclipse/che/pull/19604.

**Pull request merged**